agent/job_runner.go

```diff
@@ -582,9 +582,8 @@ func (r *JobRunner) executePreBootstrapHook(ctx context.Context, hook string) (b
 		l: r.agentLogger,
 	}
 
-	if err := sh.RunWithoutPrompt(ctx, hook); err != nil {
-		fmt.Printf("err: %s\n", err)
-		r.agentLogger.Error("Finished pre-bootstrap hook %q: job rejected", hook)
+	if err := sh.RunScript(ctx, hook, nil); err != nil {
```

Comment:
This can actually run binary hooks as well, it falls back to just trying to execute the file if it can't detect the script type: https://github.com/buildkite/agent/blob/4a98b41cac8dc9e9b811c89c51565459dba1b757/internal/job/shell/shell.go#L434-L437